tools/benchmark_tool/setup.py

```diff
@@ -67,7 +70,14 @@ def read_requirements(path: str) -> List[str]:
         # get rid of newlines
         line = line.replace('\n', '')
         # if version is specified (non-word chars present)
-        if re.search('(~|=|<|>|;)', line):
+        package_constraint = constraints.get(line.split(';')[0])
+        if re.search('(~|=|<|>)', line):
+            if package_constraint:  # both markers and versions specified
+                marker_index = line.find(";")
```

Comment:
this code does not seem to handle a case `importlib-metadata; python_version < "3.8" and sys_platform == "win32"`.
That is because `package_constraint = constraints.get(line.split(';')[0])` will return non-empty even for `importlib-metadata` without meta.
Please check.
You seem to need a check a length of `line.split(';')` greater 1.

Reply:
That's a great catch, I fixed it. Thank you.